src/report.rs

```diff
@@ -26,7 +26,12 @@ pub struct Report {
 
 impl Report {
   /// Create a new instance.
-  pub fn new(method: Method, explanation: String) -> Self {
+  pub fn new(
+    name: &str,
+    version: &str,
+    method: Method,
+    explanation: String,
```

Comment:
The String here surprises me. It's not wrong, but… I might start a Cow-olution soon!